source/game.popularity.person.bmx

```diff
@@ -115,11 +116,18 @@ Type TGameModifierPopularity_ModifyPersonPopularity extends TGameModifierPopular
 
 	Method Init:TGameModifierPopularity_ModifyPersonPopularity(data:TData, extra:TData=null)
 		Super.Init(data, extra)
+
+		If data And data.GetString("time") Then self.data = data.copy()
+
 		Return self
 	End Method
 
 
 	Method GetPopularity:TPopularity() override
+		Local guidBackup:String = popularityReferenceGUID
+		If popularityReferenceGUID And popularityReferenceGUID.contains("${")
+			popularityReferenceGUID = GameScriptExpression.ParseLocalizedText(popularityReferenceGUID, new SScriptExpressionContext(null, 0, passedParams.get("variables"))).ToString()
```

Comment:
Bei solchen "indirekt machbar-Machungen" weiss ich nicht, ob wir ueberhaupt die GUID brauchen, oder ob hier nicht die ID reicht?
Ausnahme waerend ja Scriptexpressions die eine "GUID" zusammensetzen (wenn die bspweise von einem Autoren nach gewissem Schema aufgebaut sind).
Falls eh nur "ID" statt GUID, brauchen wir "popularityReferenceGUID" nicht.

Reply:
Die IDs hat man in der Datenbank aber nicht, die GUIDs schon.
Die Idee war hier, für den personPopularity-Modifier keine feste GUID (existierendes und verpflichtendes Feld aus der Datenbank für diesen Effekt) mitzugeben, sondern eine Variable, die eine Alternative von 3 GUIDs definiert.